EAR-1214-build-toggle-switches-for-each-theme

MattElcock opened the pull request:
### What is the context of this PR?

This PR builds a new collapsible component that is opened and closed by a toggle switch, and accepts content to show when open as well as extra content to display in the header (facilitating the later addition of the preview themes radio).

### How to review

Check that the ACs are covered:

Given I'm on the Themes tab of the Settings page
then I see the title of the different themes (GB, NI, COVID, EPE, EPE NI, UKIS, UKIS NI)
and a toggle switch next to each theme

Given I'm on the Themes tab of the Settings page
then the GB theme is turned on by default
and all of the other themes are turned off by default

Given I'm on the Themes tab of the Settings page
when a theme is toggled on
then the space underneath the theme expands but is empty (eQ ID etc will be added as further AC)

- Import the **Capability examples** questionnaire from pre-prod Author into your local environment and:
  - ensure it can be opened in Author;
  - then, ensure it can be viewed in Runner by pressing the **view survey** button
- Does this require a migration? We need one if existing JSON schema properties change

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process

zimboflyman reviewed (changes requested):
Nice - just 2 things:
1) AC says there should  be a toggle for UKIS NI?
2) more of a design thing - but might look a lot tidier if the toggle switches were all lined up?

MattElcock commented:
> Nice - just 2 things:
> 
> 1. AC says there should  be a toggle for UKIS NI?
> 2. more of a design thing - but might look a lot tidier if the toggle switches were all lined up?

I've fixed #1 and shot a message to Joe about #2.

MattElcock commented:
> > Nice - just 2 things:
> > 
> > 1. AC says there should  be a toggle for UKIS NI?
> > 2. more of a design thing - but might look a lot tidier if the toggle switches were all lined up?
> 
> I've fixed #1 and shot a message to Joe about #2.

From Joe:

> Hi mate, not, thats why we removed those dividing vertical lines, if the toggles were lined up then the gaps would vary as some text is longer than the others so the gaps would all be different

tjamesmac commented:
The storybook story has a `.js` file extension instead of a `.mdx`

tjamesmac commented:
To solve the gaps issue with the toggles, could we like extract the word theme from each toggle and have them under a header? Then we could remove theme and we could have a set distance from the theme name and all the toggles would be aligned. I know it's not up to you. They do look kinda mad all over the place

MattElcock commented:
> The storybook story has a `.js` file extension instead of a `.mdx`

That's intentional, is there a problem with using the JS files instead of MDX? I realise it's not consistent, but I found it easier to write after I tried it with the comments panel refactoring story.

MattElcock commented:
> To solve the gaps issue with the toggles, could we like extract the word theme from each toggle and have them under a header? Then we could remove theme and we could have a set distance from the theme name and all the toggles would be aligned. I know it's not up to you. They do look kinda mad all over the place



> The storybook story has a `.js` file extension instead of a `.mdx`


Spoken about over Slack; all is well in the realm of Author PRs.

tjamesmac commented on eq-author/src/App/settings/ThemesPage.js:
```diff
@@ -42,6 +43,25 @@ const StyledPanel = styled.div`
+  { title: "UKIS theme" },
+  { title: "UKIS NI theme" },
+];
+
```
Not here specifically but on line 68, there's a hook called `useRouteMatch` that's used to get the params. Could we use useParams instead? We are only getting the params from it

tjamesmac commented on eq-author/src/components/CollapsibleToggled/index.js:
```diff
@@ -0,0 +1,73 @@
+  }
+`;
+const Body = styled.div`
+  display: ${(props) => (props.isOpen ? "block" : "none")};
```
```suggestion

```

tjamesmac commented on eq-author/src/components/CollapsibleToggled/index.js:
```diff
@@ -0,0 +1,73 @@
+        />
+        {headerContent}
+      </Header>
+      <Body isOpen={isOpen} data-test="CollapsibleToggled__Body">
+        {children}
+      </Body>
```
```suggestion
      {isOpen && <Body data-test="CollapsibleToggled__Body">{children}</Body>}
```

Not sure how you feel about this instead, it's a bit more obvious that its rendered conditionally instead of being in the css

MattElcock replied:
Sure.

tjamesmac commented:
Looks good, there are some accessibility issues raised by storybook for the collapsible toggled component. Not sure if you want to amend those now?

Happy to tick otherwise
